src/operator/subgraph/partitioner/custom_subgraph_property.h

```diff
@@ -0,0 +1,235 @@
+    // loop and add node names for each supported node ID
+    for (unsigned i = 0; i < supportedNodeIDs.size(); i++) {
+      if (supportedNodeIDs[i]) {
+        supportedNodes.insert(idx[i].source->attrs.name);
+      }
+    }
+  }
+  // override CreateSubgraphNode
+  virtual nnvm::NodePtr CreateSubgraphNode(const nnvm::Symbol &sym,
+                                           const int subgraph_id = 0) const {
+    int accept = 1;
+    int num_attr = 0;
+    char** attr_keys = nullptr;
+    char** attr_vals = nullptr;
+    if (acceptSubgraph_) {
+      nnvm::Graph g;
+      g.outputs = sym.outputs;
+      const auto& idx = g.indexed_graph();
+
+      // set isArg/isAux for each null op/param in the graph
+      const std::vector<std::string> aux_names = sym.ListInputNames(nnvm::Symbol::kAuxiliaryStates);
+      std::unordered_set<std::string> aux_set(aux_names.begin(), aux_names.end());
+      for (unsigned i = 0; i < idx.num_nodes(); i++) {
+        nnvm::Node* node = const_cast<nnvm::Node*>(idx[i].source);
+        // check if this node is input to subgraph
+        if (node->is_variable()) {
+          // check if this node is an aux param
+          if (aux_set.count(node->attrs.name))
+            node->attrs.dict["isAux"] = "True";
+          else
+            node->attrs.dict["isAux"] = "False";
+        }
+      }
+
+      std::string subgraph_json = nnvm::pass::SaveJSON(g);
+      CHECK(callAcceptSubgraph_(acceptSubgraph_, subgraph_json.c_str(),
+                                subgraph_id, &accept, opt_keys_.data(),
+                                opt_vals_.data(), opt_keys_.size(),
+                                &attr_keys, &attr_vals, &num_attr))
+        << "Error calling acceptSubgraph for '" << subgraphProp << "'";
+    }
+    if (accept) {
+      nnvm::NodePtr n = nnvm::Node::Create();
+      n->attrs.op = Op::Get(subgraph_op_name);
+      n->attrs.name = "_op" + std::to_string(subgraph_id);
+      n->attrs.subgraphs.push_back(std::make_shared<nnvm::Symbol>(sym));
+      // set user specified attributes
+      for (int i=0; i < num_attr; i++) {
+        n->attrs.dict[attr_keys[i]] = attr_vals[i];
+        callFree_(attr_vals[i]);
+        callFree_(attr_keys[i]);
+      }
+      // free memory used by custom op to allocate attributes
+      callFree_(attr_vals);
+      callFree_(attr_keys);
+      return n;
+    } else {
+      return NULL;
+    }
+  }
+  // override CreateSubgraphSelector
+  virtual SubgraphSelectorPtr CreateSubgraphSelector() const {
+    return std::make_shared<CustomContainOpSelector>(supportedNodes);
+  }
+
+  std::string subgraphProp;
+  partCallSupportedOps_t callSupportedOps_;
+  supportedOps_t supportedOps_;
+  partCallAcceptSubgraph_t callAcceptSubgraph_;
+  acceptSubgraph_t acceptSubgraph_;
+  opCallFree_t callFree_;
+  std::unordered_set<std::string> supportedNodes;
+  std::string subgraph_op_name;
+  std::vector<std::pair<std::string, std::string>> options_map_;
```

Comment:
there is `supportedNodes` and `options_map_`. 
I don't want to nick-pick too much here but shall we have consistent naming for the variable names? shall we use snake_case_ consistently?

Reply:
`supportedNodes` is a function name and `options_map` is a variable name, not a function name. we should have separate rules for each, or vote on dev-mailing-list for a single standard for all naming (function, variable, etc).

Reply:
For this PR, it looks like `supportedNodes` is a variable instead. Also, at least for `subgraph_op_name`, it should have an underscore suffix. 

For MXNet in general, I'll start a thread for the coding style/naming standard on dev@

Reply:
sorry. you're right, ive made the changed.